### ISSUE TITLE
1.7.10 Incorrect Player Count

### ISSUE THREAD

**Issue by xXBuilderBXx** [ldilley/minestat#5]
All 1.7.10 servers are showing an incorrect player and max player count

Top status is using your script
Bottom one is using this https://mcapi.us/

Picture http://imgur.com/a/sJZcP
Im using c#

**Comment by ldilley** [comment]
Thank you for the report, Builderb. Did this symptom appear after the latest pull request merger/are you using the latest commit? I am wondering if this broke before or after the last commit.

**Comment by xXBuilderBXx** [comment]
Yes i used the latest one

**Comment by ldilley** [comment]
I am not able to reproduce this symptom using the latest commit.  Not only did I try setting up a test 1.7.10 vanilla server, but I also pointed the old and new Example.cs to your server. All 3 methods are working as intended. See below for details.

`
[lloyd@lindev]$ mcs Example2.cs MineStat.cs
[lloyd@lindev$ mono Example2.exe
Minecraft server status of localhost on port 25565:
Server is online running version 1.7.10 with 2 out of 20 players.
Message of the day: A Minecraft Server
`

Example2.cs contains the latest C# example from README.md (with the missing parenthesis fixed.) The server is using a vanilla 1.7.10 jar from minecraft.net and is hosted on my LAN. After adding several connections, the above player account increments as expected. I'm using Mono 4.4.0 to compile and run the Example and Example 2 class.

I also used both the legacy Example.cs and the new code below with your FQDN plugged into the connect method. Below are the results.

`
[lloyd@lindev]$ cat Example2.cs
using System;

class Example2
{
  public static void Main()
  {
    MineStat ms = new MineStat("tl.devcoftb.eu", 25565);
    Console.WriteLine("Minecraft server status of {0} on port {1}:", ms.Address, ms.Port);
    if(ms.ServerUp)
    {
      Console.WriteLine("Server is online running version {0} with {1} out of {2} players.", ms.Version, ms.CurrentPlayers, ms.MaximumPlayers);
      Console.WriteLine("Message of the day: {0}", ms.Motd);
    }
    else
      Console.WriteLine("Server is offline!");
  }
}
[lloyd@lindev]$ mcs Example2.cs MineStat.cs
[lloyd@lindev]$ mono Example2.exe
Minecraft server status of tl.devcoftb.eu on port 25565:
Server is online running version BungeeCord 1.8 with 128 out of 500 players.
Message of the day: §aDeVco Servers §bEU
`

Of course, the server you referenced is showing 1.8. However, both 1.7.10 and your version seem to be working as intended. Is it possible to see a snippet of your implementation? If not, it may have something to do with the legacy/obsolete class code contributed by @kolletzki.

**Comment by xXBuilderBXx** [comment]
Actually the player count is not 128/500 it should be /50 that is the issue also it might be possible that bungee cord is the reason it is showing that so it must be correct then >_<

**Comment by kolletzki** [comment]
Dumping the `rawServerData` object in LinqPad gave me the following output:

![](https://i.imgur.com/OsN823j.png)

The first marked section represents the current player count as a string:
```
dec 53 = utf8 "5"
dec 51 = utf8 "3"
```

The second section represents the max player count:
```
dec 53 = utf8 "5"
dec 48 = utf8 "0"
dec 48 = utf8 "0"
```

I ran this with both, the "old" and the "new" version of MineStat.cs, both with the same result. Running the request with a TCP client app also delivered this response.

The way MineStat fetches the player stats seems to be incompatible with BungeeCord/your server.
If I find some time, I'll try to dig into this.

**Comment by xXBuilderBXx** [comment]
Ok cool no problem glad to see that you are still active with this ;)

**Comment by ldilley** [comment]
I've reviewed the BungeeCord source code responsible for supplying this information and see nothing that would append an additional zero. The specific BungeeCord module that handles this data if you're interested is: https://github.com/SpigotMC/BungeeCord/blob/master/query/src/main/java/net/md_5/bungee/query/QueryHandler.java

What I *think* is occurring here is:
1.) The protocol is indeed behaving properly between MineStat and BungieCord.
2.) There is some confusion regarding what values you are expecting to see back from the server. BungieCord is a proxy service that sits between clients and Minecraft servers. You are expecting to see the "max-players" value defined in server.properties for a server instance (Spigot, vanilla, Cuberite, Sponge, or otherwise.) What you are actually receiving back is either the global "player_limit" or a listener's "max_players" value from BungieCord. These two BungieCord values can differ from one another and also differ from the actual Minecraft server value.

In other words, please review your BungieCord's config.yml and check for the above two parameters. I am guessing one of them is actually set to "500". ;)

**Comment by xXBuilderBXx** [comment]
Ah i dont have access to the server its just that i made an uptime bot on discord for the minecraft network thanks for the reply anyway.

**Comment by ldilley** [comment]
Don't lose hope, BuilderB. While I've confirmed that BungeeCord indeed sends these:

`writeNumber( out, listener.getMaxPlayers() );`

or

`data.put( "maxplayers", Integer.toString( listener.getMaxPlayers() ) );`

(which means you are getting a listener's max-players value from BungeeCord's config.yml and **NOT** the max_players value from an actual Minecraft server sitting behind BungeeCord), you can still use the BungeeCord API to query a server being reverse proxied by BungeeCord. See this page of the API specifically: https://ci.md-5.net/job/BungeeCord/ws/api/target/apidocs/net/md_5/bungee/api/connection/Server.html

It looks like arbitrary data can be sent via the sendData() method. This _may_ get you what you need. Unfortunately, it does not look like the ServerInfo class has a method to obtain the player capacity for some reason... only current player count by way of getting the size of a returned collection.

The BungeeCord Plugin Messaging Channel may also be of interest to you: https://www.spigotmc.org/wiki/bukkit-bungee-plugin-messaging-channel/

Lastly, it might be an option to add the functionality to MineStat to validate whether the front-end server is running BungeeCord. If so, send certain data to potentially get at the inner server max count. We would have to take a closer look at the BungeeCord source code to get a better idea of what to send and expect then see if it is a possibility from there.

I'll close this issue since things are technically behaving as intended. I'll convert it to a feature request instead and see if I or someone else can pick it up as time allows.

**Comment by xXBuilderBXx** [comment]
Glad to see that you are dedicated to this project nice work dude.
